feature_engine/prediction/target_mean_classifier.py

```diff
@@ -0,0 +1,195 @@
+    For numerical variables, the values are first sorted into bins of equal-width or
+    equal-frequency. Then, the mean target value is estimated for each bin. If the
+    variables are categorical, the mean target value is estimated for each category.
+    Finally, the estimator takes the average of the mean target value across the
+    input variables and determines the class based on a threshold of 0.5.
+
+    Parameters
+    ----------
+    variables: list, default=None
+        The list of input variables. If None, the estimator will use all variables as
+        input features (except datetime).
+
+    bins: int, default=5
+        If the dataset contains numerical variables, the number of bins into which
+        the values will be sorted.
+
+    strategy: str, default='equal_width'
+        Whether the bins should be of equal width ('equal_width') or equal frequency
+        ('equal_frequency').
+
+    Attributes
+    ----------
+    variables_categorical_:
+        The group of categorical input variables that will be used for prediction.
+
+    variables_numerical_:
+        The group of numerical input variables that will be used for prediction.
+
+    classes_:
+        A list of class labels known to the classifier.
+
+    pipeline_:
+        A Scikit-learn Pipeline with a dicretiser and/or encoder. Used to determine the
+        mean target value per category or bin, per variable.
+
+    n_features_in_:
+        The number of features in the train set used in fit.
+
+    Methods
+    -------
+    fit:
+        Learn the mean target value per category or per bin, for each variable.
+
+    predict:
+        Predict class labels for samples in X.
+
+    predict_log_proba:
+        Predict logarithm of probability estimates.
+
+    predict_proba:
+        Proxy for Probability estimates based of the average of the target mean value
+         across variables.
+
+    score:
+        Return the mean accuracy on the given test data and labels.
+
+    See Also
+    --------
+    feature_engine.encoding.MeanEncoder
+    feature_engine.discretisation.EqualWidthDiscretiser
+    feature_engine.discretisation.EqualFrequencyDiscretiser
+
+    References
+    ----------
+    Adapted from:
+
+    .. [1] Miller, et al. "Predicting customer behaviour: The University of Melbourne’s
+        KDD Cup report". JMLR Workshop and Conference Proceeding. KDD 2009
+        http://proceedings.mlr.press/v7/miller09/miller09.pdf
+    """
+
+    def fit(self, X: pd.DataFrame, y: pd.Series):
+        """
+        Learn the mean target value per category or bin.
+
+        Parameters
+        ----------
+        X : pandas dataframe of shape = [n_samples, n_features]
+            The training input samples.
+
+        y : pandas series of shape = [n_samples,]
+            The target variable.
+        """
+        # check that y is binary
+        self.classes_ = list(y.unique())
```

Comment:
can we use unique_labels from sklearn instead?
https://scikit-learn.org/stable/developers/develop.html#rolling-your-own-estimator